startup/buildInfo.js

```diff
@@ -179,20 +185,76 @@ if (builderType === "paketo") {
         };
     }
 
-    imageBuildProps = {
-        appId,
+    return {
         config: {
             type: "nixpacks",
             version: nixpacksVersion,
             plan,
         },
-        env: userEnv,
+        env: {},
+    };
+}
+
+/**
+ * @returns {CreateBuild}
+ */
+function makeBuildProps() {
+    /**
+     * @type {import("../common").Config}
+     */
+    const appConfig = loadAdaptableAppConfig();
+
+    const appId = process.env.ADAPTABLE_APP_ID;
+    if (appId == null) throw new Error("No ADAPTABLE_APP_ID found");
+
+    const revId = process.env.ADAPTABLE_APPREVISION_ID;
+    if (revId == null) throw new Error("No ADAPTABLE_APPREVISION_ID");
+
+    const tags = (process.env.ADAPTABLE_TEMPLATE_TAGS || "").split(",");
+
+    let { builderType } = appConfig;
+
+    const userEnv = appConfig.buildEnvironment || {};
+
+    if (tags.includes("go") || tags.includes("php")) {
+        if (builderType === "paketo") {
+            const msg = `${tags.join("+")} not currently supported with paketo builderType. Use nixpacks instead.`;
+            // eslint-disable-next-line no-console
+            console.log(msg);
+            throw new Error(msg);
+        }
+        builderType = "nixpacks";
+    } else if (tags.includes("dockerfile")) {
+        builderType = "dockerfile";
+    } else if (!builderType) {
+        builderType = "paketo";
+    }
+
+    /**
+     * @type {BuildInfo}
+     */
+    // eslint-disable-next-line no-nested-ternary
```

Comment:
Any way to use a different construct here besides nested ternary?  Maybe pull out into a function? Otherwise, I prefer a switch case in an IIFE rather than this.  But, if you prefer this to the other options, it isn't worth holding things up.

Reply:
Fixed